benchmarking/runs/priority_algorithm/diversify_gender_min_2/max_spread.py

```diff
@@ -0,0 +1,243 @@
+            "PrioritySatisfaction": PrioritySatisfaction(
+                goals_to_priorities(scenario.goals),
+                False,
+            ),
+        }
+
+        artifacts: Dict[int, SimulationSetArtifact] = {}
+
+        for spread in max_spread:
+            print("MAX SPREAD /", spread)
+            # set up either mock or real data
+            student_provider_settings = MockStudentProviderSettings(
+                number_of_students=number_of_students,
+                attribute_ranges={
+                    ScenarioAttribute.GENDER.value: [
+                        (Gender.MALE, 1 - ratio_of_female_students),
+                        (Gender.FEMALE, ratio_of_female_students),
+                    ],
+                },
+            )
+
+            simulation_set_artifact = SimulationSet(
+                settings=SimulationSettings(
+                    num_teams=number_of_teams,
+                    scenario=scenario,
+                    student_provider=MockStudentProvider(student_provider_settings),
+                    cache_key=f"priority_algorithm/diversify_gender_min_2/max_spread/{spread}",
+                ),
+                algorithm_set={
+                    AlgorithmType.RANDOM: [RandomAlgorithmConfig()],
+                    AlgorithmType.WEIGHT: [WeightAlgorithmConfig()],
+                    AlgorithmType.PRIORITY: [
+                        PriorityAlgorithmConfig(
+                            MAX_ITERATE=max_iterate,
+                            MAX_TIME=max_time,
+                            MAX_KEEP=max_keep,
+                            MAX_SPREAD=spread,
+                        ),
+                        PriorityAlgorithmConfig(
+                            name="local_max",
+                            MUTATIONS=[
+                                (mutate_local_max, 1),
+                                (mutate_random_swap, spread - 1),
+                            ],
+                            MAX_TIME=max_time,
+                            MAX_ITERATE=max_iterate,
+                            MAX_KEEP=max_keep,
+                            MAX_SPREAD=spread,
+                        ),
+                        PriorityAlgorithmConfig(
+                            name="local_max_random",
+                            MUTATIONS=[
+                                (mutate_local_max_random, 1),
+                                (mutate_random_swap, spread - 1),
+                            ],
+                            MAX_TIME=max_time,
+                            MAX_ITERATE=max_iterate,
+                            MAX_KEEP=max_keep,
+                            MAX_SPREAD=spread,
+                        ),
+                        PriorityAlgorithmConfig(
+                            name="local_max_double_random",
+                            MUTATIONS=[
+                                (mutate_local_max_double_random, 1),
```

Comment:
things get weird here because if we have max spread of 10, for example, why only run the local max double random once? Why not 5 times?